### PR TITLE
BZ2033449: Configuring FIFO priority scheduling for PTP hardware

### PR DESCRIPTION
**Pull request by aireilly**
Documents https://bugzilla.redhat.com/show_bug.cgi?id=2033449 This work is for main, enterprise-4.9, enterprise-4.10

Preview: https://deploy-preview-39936--osdocs.netlify.app/openshift-enterprise/latest/networking/using-ptp#cnf-configuring-fifo-priority-scheduling-for-ptp_using-ptp

### CODE REVIEW

File: modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+----
+$ oc -n openshift-ptp logs linuxptp-daemon-lgm55 -c linuxptp-daemon-container|grep chrt
+----
++
+.Example output
+[source,terminal]
+----
+...
+I1216 19:24:57.091872 1600715 daemon.go:285] /bin/chrt -f 65 /usr/sbin/ptp4l -f /var/run/ptp4l.0.config -2  --summary_interval -4 -m
+...
```
Comment (bobfuru) [inline modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc:74]: Wondering if the ellipses above and below are really necessary? I suspect that more output is returned and that you want the user to only pay attention to this output snippet, so probably fine as you have it.

File: modules/nw-ptp-configuring-linuxptp-services-as-boundary-clock.adoc [hunk 2]
```diff
@@ -153,12 +155,14 @@ spec:
 <7> The interface name to synchronize from.
 <8> The interface to synchronize devices connected to the interface.
 <9> Specify system config options for the `phc2sys` service, for example `-a -r`. If this field is empty the PTP Operator does not start the `phc2sys` service.
-<10> Specify an array of one or more `recommend` objects which define rules on how the `profile` should be applied to nodes.
-<11> Specify the `profile` object name defined in the `profile` section.
-<12> Specify the `priority` with an integer value between `0` and `99`. A larger number gets lower priority, so a priority of `99` is lower than a priority of `10`. If a node can be matched with multiple profiles according to rules defined in the `match` field, the profile with the higher priority is applied to that node.
-<13> Specify `match` rules with `nodeLabel` or `nodeName`.
-<14> Specify `nodeLabel` with the `key` of `node.Labels` from the node object.
-<15> Specify `nodeName` with `node.Name` from the node object.
+<10> Scheduling policy for ptp4l and phc2sys processes. Default value is `SCHED_OTHER`. Use `SCHED_FIFO` on systems that support FIFO scheduling.
```
Comment (bobfuru) [inline modules/nw-ptp-configuring-linuxptp-services-as-boundary-clock.adoc:158]: I suggest using backticks for the process names here (`ptp4l` and `phc2sys`).

File: modules/nw-ptp-configuring-linuxptp-services-as-boundary-clock.adoc [hunk 3]
```diff
@@ -153,12 +155,14 @@ spec:
...
+<11> Integer value from 1-65 used to set FIFO priority for `ptp4l` and `phc2sys` processes. Required if `SCHED_FIFO` is set for `ptpSchedulingPolicy`.
+<12> Specify an array of one or more `recommend` objects which define rules on how the `profile` should be applied to nodes.
```
Comment (bobfuru) [inline modules/nw-ptp-configuring-linuxptp-services-as-boundary-clock.adoc:160]: ISG: Use "which", preceded by a comma, to introduce a nonrestrictive clause. Use "that", without a comma, to introduce a restrictive clause.
```suggestion
<12> Specify an array of one or more `recommend` objects that define rules on how the `profile` object should be applied to nodes.
```

File: modules/nw-ptp-configuring-linuxptp-services-as-ordinary-clock.adoc [hunk 4]
```diff
@@ -46,12 +48,14 @@ spec:
...
 <6> Specify system config options for the `phc2sys` service, for example `-a -r`. If this field is empty the PTP Operator does not start the `phc2sys` service.
 <7> Specify a string that contains the configuration to replace the default `/etc/ptp4l.conf` file. To use the default configuration, leave the field empty.
-<8> Specify an array of one or more `recommend` objects which define rules on how the `profile` should be applied to nodes.
-<9> Specify the `profile` object name defined in the `profile` section.
-<10> Specify the `priority` with an integer value between `0` and `99`. A larger number gets lower priority, so a priority of `99` is lower than a priority of `10`. If a node can be matched with multiple profiles according to rules defined in the `match` field, the profile with the higher priority is applied to that node.
-<11> Specify `match` rules with `nodeLabel` or `nodeName`.
-<12> Specify `nodeLabel` with the `key` of `node.Labels` from the node object.
-<13> Specify `nodeName` with `node.Name` from the node object.
+<8> Scheduling policy for ptp4l and phc2sys processes. Default value is `SCHED_OTHER`. Use `SCHED_FIFO` on systems that support FIFO scheduling.
```
Comment (bobfuru) [inline modules/nw-ptp-configuring-linuxptp-services-as-ordinary-clock.adoc:51]: Same as previous comment about backticks.

File: modules/nw-ptp-configuring-linuxptp-services-as-ordinary-clock.adoc [hunk 5]
```diff
@@ -46,12 +48,14 @@ spec:
...
+<9> Integer value from 1-65 used to set FIFO priority for `ptp4l` and `phc2sys` processes. Required if `SCHED_FIFO` is set for `ptpSchedulingPolicy`.
+<10> Specify an array of one or more `recommend` objects which define rules on how the `profile` should be applied to nodes.
```
Comment (bobfuru) [inline modules/nw-ptp-configuring-linuxptp-services-as-ordinary-clock.adoc:53]: s/which/that

**Review by bobfuru — COMMENTED** [review]
Great work! Just a few small nits. Please be sure to show commits when ready for merge. Also, I'm applying the 4.10 label as wells, as I suspect this version applies as well if merging to main.

**Comment by aireilly** [comment]
LGTM from @obochan-rh 2021-12-29 on [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=2033449)

**Comment by aireilly** [comment]
> Great work! Just a few small nits. Please be sure to show commits when ready for merge. Also, I'm applying the 4.10 label as wells, as I suspect this version applies as well if merging to main.

Thanks! implmented your changes Bob.

File: modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  namespace: openshift-ptp
+...
+spec:
+  profile:
+  - name: "profile1"
+...
+    ptpSchedulingPolicy: SCHED_FIFO <1>
+    ptpSchedulingPriority: 65 <2>
+----
+<1> Scheduling policy for ptp4l and phc2sys processes. Use `SCHED_FIFO` on systems that support FIFO scheduling.
```
Comment (stevsmit) [inline modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc:43]: Throughout the rest of this document, `ptp4l` and `phc2sys` are in backtickets. In this instance (<1>), they are not. Should it be fixed?

Comment (aireilly) [reply modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc:43]: Yes, will correct.

Comment (aireilly) [reply modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc]: undid my erroneous commit and applied your suggestion, thanks.

**Pull request merged**